docs/tools/prism-eslint-hook.js

```diff
@@ -0,0 +1,179 @@
+"use strict";
+
+const Prism = require("prismjs");
+const { Linter } = require("../../lib/api");
+const astUtils = require("../../lib/shared/ast-utils");
+
+/** @typedef {import("../../lib/shared/types").ParserOptions} ParserOptions */
+
+/**
+ * A set of content that needs to be marked.
+ * @type {Map<string, ParserOptions>}
+ */
+const contentsMustBeMarked = new Map();
+
+/**
+ * Add content that needs to be marked.
+ * @param {string} content Source code content that marks eslint errors.
+ * @param {ParserOptions} options The options used for validation.
+ * @returns {void}
+ */
+function addContentMustBeMarked(content, options) {
+    contentsMustBeMarked.set(content, options);
```

Comment:
I checked and we don't have any duplicate `content`, so this is okay. If we had any duplicates with different parser options, we could not rely just on the content as a key.

Reply:
Prism.hooks only has code text in markdown, so I think it would be difficult to handle anything else as a key. For now, this is processed synchronously, so I don't think there is any problem with duplication. However, there may be problems after updating the document system 😓

Reply:
Every time execution hits this line, `contentsMustBeMarked` is empty, and every time execution hits the line with `contentsMustBeMarked.delete(env.code)` in the hook callback, the map has only one element: the content to be removed. So it seems that markdownIt is calling the highlighter immediately after calling the other plugin functions for the same block. If this is the case then we are safe: there can't be a duplication if there is just one potential match.

This makes me also think that probably we could just remember the last `content` and `options` found here instead of using a map. Do you think that would work?

Reply:
> This makes me also think that probably we could just remember the last `content` and `options` found here instead of using a map. Do you think that would work?

That's right. I think it works. I would replace Map :+1: